### PR TITLE
[#2602] [JSDoc] Add missing documentation for botbuilder-lg root files

### PR DESCRIPTION
**Pull request by denscollo**
Addresses #2602

## Description
This PR adds the missing documentation in [botbuilder-lg](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botbuilder-lg) library.
Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)
fixes.

## Specific Changes

- Added JSDoc comments in all public methods and some of the private ones in the following files:
    - analyzer
    - analyzerResult
    - customizedMemory
    - diagnostic
    - errorListener
    - evaluationOptions
    - evaluationTarget
    - evaluator
    - expander
    - extractor
    - lgResource
    - multiLanguageLG

## Testing
The following image shows the ESLint Report with no JSDoc errors for the files updated.
image

### CODE REVIEW

**Comment by chrimc62** [comment]
> 

Documentation?

---
Refers to: libraries/botbuilder-lg/src/evaluationOptions.ts:48 in f20991c. [](commit_id = f20991c7a1c049d8364b89f774a197f6e1821bf8, deletion_comment = False)

File: libraries/botbuilder-lg/src/evaluator.ts [hunk 1]
```diff
@@ -308,14 +349,29 @@ export class Evaluator extends AbstractParseTreeVisitor<any> implements LGTempla
         return undefined;
     }
 
+    /**
+     * Replaces an expression contained in a text.
```
Comment (chrimc62) [inline libraries/botbuilder-lg/src/evaluator.ts:353]: >a [](start = 43, length = 1)

remove

File: libraries/botbuilder-lg/src/evaluator.ts [hunk 2]
```diff
@@ -328,6 +384,15 @@ export class Evaluator extends AbstractParseTreeVisitor<any> implements LGTempla
         return errorMsg;
     }
 
+    /**
+     * Checks an Expression Result and throws the corresponding error.
```
Comment (chrimc62) [inline libraries/botbuilder-lg/src/evaluator.ts:388]: >Expression Result [](start = 17, length = 17)

don't need to capitalize

File: libraries/botbuilder-lg/src/expander.ts [hunk 3]
```diff
@@ -49,6 +49,12 @@ export class Expander extends AbstractParseTreeVisitor<string[]> implements LGTe
     private readonly evaluationTargetStack: EvaluationTarget[] = [];
     private readonly lgOptions: EvaluationOptions;
 
+    /**
+     * Creates a new instance of the Expander class.
+     * @param templates Template list.
+     * @param expressionParser Given expression parser.
+     * @param opt Options for LG including strictMode, replaceNull and lineBreakStyle.
```
Comment (chrimc62) [inline libraries/botbuilder-lg/src/expander.ts:56]: >including strictMode, replaceNull and lineBreakStyle. [](start = 33, length = 53)

Don't really need this here--it is in the underlying options class.

**Review by chrimc62 — CHANGES_REQUESTED** [review]
:clock1:

File: libraries/botbuilder-lg/src/analyzer.ts [hunk 4]
```diff
@@ -82,6 +95,10 @@ export class Analyzer extends AbstractParseTreeVisitor<AnalyzerResult> implement
         return result;
     }
 
+    /**
+     * Visit a parse tree produced by LGTemplateParser.structuredTemplateBody.
+     * @param ctx The parse tree.
```
Comment (joshgummersall) [inline libraries/botbuilder-lg/src/analyzer.ts:100]: It looks like many of this method docs are missing a `@returns` line.

File: libraries/botbuilder-lg/src/evaluator.ts [hunk 5]
```diff
@@ -126,6 +132,10 @@ export class Evaluator extends AbstractParseTreeVisitor<any> implements LGTempla
         return result;
     }
 
+    /**
+     * Visit a parse tree produced by LGTemplateParser.structuredTemplateBody.
+     * @param ctx The parse tree.
```
Comment (joshgummersall) [inline libraries/botbuilder-lg/src/evaluator.ts:137]: Needs `@returns`

File: libraries/botbuilder-lg/src/expander.ts [hunk 6]
```diff
@@ -90,10 +96,18 @@ export class Expander extends AbstractParseTreeVisitor<string[]> implements LGTe
         return result;
     }
 
+    /**
+     * Visit a parse tree produced by the normalBody labeled alternative in LGTemplateParser.body.
+     * @param ctx The parse tree.
```
Comment (joshgummersall) [inline libraries/botbuilder-lg/src/expander.ts:101]: `@returns`

File: libraries/botbuilder-lg/src/extractor.ts [hunk 7]
```diff
@@ -49,6 +58,10 @@ export class Extractor extends AbstractParseTreeVisitor<Map<string, any>> implem
         return result;
     }
 
+    /**
+     * Visit a parse tree produced by LGTemplateParser.normalTemplateBody.
+     * @param context The parse tree.
```
Comment (joshgummersall) [inline libraries/botbuilder-lg/src/extractor.ts:63]: `@returns`

**Review by joshgummersall — CHANGES_REQUESTED** [review]
Looks like quite a few doc comments are missing the `@returns` line. Otherwise looks good!

**Review by joshgummersall — APPROVED** [review]

**Pull request merged**